cmake/loki_transform.cmake

```diff
@@ -5,188 +5,32 @@
-        set( _LOKI_TRANSFORM loki-transform.py )
-        set( _LOKI_TRANSFORM_DEPENDENCY "" )
-    endif()
-
-endmacro()
-
-##############################################################################
-
-macro( _loki_transform_parse_convert_args _func_name )
-
-    _loki_transform_parse_args( ${_func_name} )
-
-    if( NOT _PAR_OUTPUT )
-        ecbuild_critical( "No OUTPUT specified for ${_func_name}()" )
-    endif()
-
-    if( NOT _PAR_DEPENDS )
-        ecbuild_critical( "No DEPENDS specified for ${_func_name}()" )
-    endif()
-
-    if( _PAR_CPP )
-        list( APPEND _ARGS --cpp )
-    endif()
-
-    if( _PAR_OUTPATH )
-        file( MAKE_DIRECTORY ${_PAR_OUTPATH} )
-        list( APPEND _ARGS --out-path ${_PAR_OUTPATH} )
-    endif()
-
-    if( _PAR_DEFINE )
-        if ( _PAR_DEFINITIONS )
-            ecbuild_critical( "Both DEFINITIONS and DEFINE given to ${_func_name}(): Please use DEFINITIONS only" )
-        else()
-            ecbuild_warn( "${_func_name}(): DEFINE is deprecated, please use DEFINITIONS" )
-            set( _PAR_DEFINITIONS ${_PAR_DEFINE} )
-        endif()
-    endif()
-
-    if( _PAR_DEFINITIONS )
-        foreach( _DEFINE ${_PAR_DEFINITIONS} )
-            list( APPEND _ARGS --define ${_DEFINE} )
-        endforeach()
-    endif()
-
-    if( _PAR_XMOD )
-        foreach( _XMOD ${_PAR_XMOD} )
-            file( MAKE_DIRECTORY ${XMOD_DIR} )
-            list( APPEND _ARGS --xmod ${_XMOD} )
-        endforeach()
-    endif()
-
-endmacro()
+include( loki_transform_helpers )
 
 ##############################################################################
 # .rst:
 #
-# loki_transform_convert
-# ======================
+# loki_transform
+# ==============
 #
-# Apply Loki transformation in convert mode.::
+# Invoke loki-transform.py using the given options.::
 #
-#   loki_transform_convert(
+#   loki_transform(
+#       COMMAND <convert|transpile|...>
 #       OUTPUT <outfile1> [<outfile2> ...]
 #       DEPENDS <dependency1> [<dependency2> ...]
 #       MODE <mode>
-#       FRONTEND <frontend>
-#       [CPP]
+#       CONFIG <config-file>
 #       [DIRECTIVE <directive>]
-#       [CONFIG <config-file>]
-#       [PATH <path>]
-#       [OUTPATH <outpath>]
-#       [INCLUDES <include1> [<include2> ...]]
+#       [CPP]
+#       [FRONTEND <frontend>]
+#       [INLINE_MEMBERS]
+#       [BUILDDIR <build-path>]
+#       [SOURCES <source1> [<source2> ...]]
 #       [HEADERS <header1> [<header2> ...]]
-#       [DEFINITIONS <define1> [<define2> ...]]
-#       [OMNI_INCLUDE <omni-inc1> [<omni-inc2> ...]]
-#       [XMOD <xmod-dir1> [<xmod-dir2> ...]]
-#       [REMOVE_OPENMP] [DATA_OFFLOAD] [GLOBAL_VAR_OFFLOAD]
-#       [TRIM_VECTOR_SECTIONS] [REMOVE_DERIVED_ARGS]
-#       [INLINE_MEMBERS] [DERIVE_ARGUMENT_ARRAY_SHAPE]
 #   )
 #
-# Call ``loki-transform.py convert ...`` with the provided arguments.
+# Call ``loki-transform.py <ecphys|...> ...`` with the provided arguments.
```

Comment:
I think ecphys is basically deprecated now too, no?

Reply:
Thanks, good spot. Yeah, ecphys is already gone. Will change to `convert`.